src/emu.h

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+
+using byte_t = uint8_t;
+using word_t = uint16_t;
+template<typename T> bool IsPtrAligned(word_t ptr);
+template<> bool IsPtrAligned<byte_t>(word_t ptr) { return true; }
+//template<> bool IsPtrAlgined<word_t>(word_t ptr) { return !(ptr % sizeof(word_t)); }
+
+struct Emu {
+	enum struct TrapVec : byte_t {
+		USER_DEV	= 0000,
+		ODD_ADDR	= 0004,
+		ILL_INSTR	= 0004,
+		STK_RED		= 0004,
+		RES_INSTR	= 0010,
+		TRACE		= 0014,
+		POWER_FAIL	= 0024,
+		EMT		= 0030,
+		TRAP		= 0034,
+		MM_TRAP		= 0250,
+	} trapVec;
+	bool trapPending = false;
+	void RaiseTrap(TrapVec const trap) { trapVec = trap; trapPending = true; };
+
+	enum struct GenRegId : byte_t {
+		R0	= 00,
+		R1	= 01,
+		R2	= 02,
+		R3	= 03,
+		R4	= 04,
+		R5	= 05,
+		R6	= 06,
+		R7	= 07,
+		PC	= R7,
+		HW_SP	= R6,
+		MAX	= 07,
+	};
+	word_t genRegfile[static_cast<byte_t>(GenRegId::MAX)];
+
+	struct Memory {
+		Emu &emu;
+		byte_t *dummy;
+		word_t const dummy_start = 00377;
```

Comment:
`// Range 000000 to 000377 is dedicated to interrupt and trap vectors(see

Reply:
start = 338, resolved

Reply:
00400, I'm sorry again.